src/search_space.h

```diff
@@ -93,7 +93,7 @@ struct SearchSpace
     Map<Node> node_map;
 
     /// @brief A map of weights corresponding to elements in @ref node_map, used to weight probabilities of each node being sampled from the map. 
-    Map<float> weight_map; 
+    Map<float> node_weights; 
```

Comment:
can we go with `node_map_weights`? node weights sounds like the weights of a node rather than a map containing weights corresponding to nodes.